chains/tendermint/query.go

```diff
@@ -249,25 +238,125 @@ func (c *Chain) QueryUnrecievedAcknowledgements(ctx core.QueryContext, seqs []ui
 	return res.Sequences, nil
 }
 
-func (src *Chain) QueryPacket(ctx core.QueryContext, seq uint64) (*chantypes.Packet, error) {
-	txs, err := src.QueryTxs(int64(ctx.Height().GetRevisionHeight()), 1, 1000, rcvPacketQuery(src.Path().ChannelID, int(seq)))
+func (c *Chain) QueryUnfinalizedRelayAcknowledgements(ctx core.QueryContext, counterparty core.LightClientICS04Querier) (core.PacketInfoList, error) {
+	res, err := c.queryPacketAcknowledgementCommitments(ctx, 0, 1000)
+	if err != nil {
+		return nil, err
+	}
+
+	var packets core.PacketInfoList
+	for _, ps := range res.Acknowledgements {
+		packet, rpHeight, err := c.queryReceivedPacket(ctx, ps.Sequence)
+		if err != nil {
+			return nil, err
+		}
+		ack, waHeight, err := c.queryWrittenAcknowledgement(ctx, ps.Sequence)
+		if err != nil {
+			return nil, err
+		}
+		var height clienttypes.Height
+		if rpHeight.LT(waHeight) {
```

Comment:
Can we rewrite the following?
```
packets = append(packets, &core.PacketInfo{
	Packet:          *packet,
	Acknowledgement: ack,
	EventHeight:     rpHeight,
})
```
Considering the order of receive-packet and write-acknowledgment, `rpHeight <= waHeight` always holds.
Therefore, if `min(rpHeight, waHeight)` is assigned to `EventHeight`, then `EventHeight` will always be equal to `rpHeight`. Do I misunderstand something?

Reply:
Fixed: https://github.com/hyperledger-labs/yui-relayer/pull/90/commits/9980e1375724fad9397ced3dce0fc855fd894e54